rest_framework/validators.py

```diff
@@ -159,17 +159,18 @@ def __call__(self, attrs, serializer):
         queryset = self.filter_queryset(attrs, queryset, serializer)
         queryset = self.exclude_current_instance(attrs, queryset, serializer.instance)
 
+        checked_names = [
+            serializer.fields[field_name].source for field_name in self.fields
```

Comment:
Since fields is a dict it is more readable to iterate over values, also it should be a little faster.

```suggestion
            field.source for field in self.fields.values()
```

Reply:
Hi @sevdog thanks for the comment, but I believe `self.fields` is a list of strings, not dict.

Reply:
Yeah, sorry I was considering to be in an other class/file.